src/pat-display-time.js

```diff
@@ -1,58 +1,59 @@
 (function (root, factory) {
-    if (typeof define === 'function' && define.amd) {
+    // We use AMD (Asynchronous Module Definition) or browser globals to create
+    // this module.
+    if (typeof define === "function" && define.amd) {
         define([
-            'pat-registry',
-            'pat-parser',
-            'moment'
-            ], function() {
-                return factory.apply(this, arguments);
-            });
-        } else {
-            factory(root.patterns, root.patterns.Parser);
-        }
-}(this, function(registry, Parser, moment) {
-    'use strict';
+            "jquery",
+            "pat-base",
+            "pat-registry",
+            "pat-parser",
+            "pat-logger",
+            "moment"
+        ], function() {
+            return factory.apply(this, arguments);
+        });
+    } else {
+        // If require.js is not available, you'll need to make sure that these
+        // global variables are available.
+        factory($, patterns.Base, patterns, patterns.Parser, patterns.logger);
+    }
+}(this, function($, Base, registry, Parser, logger, moment) {
+    "use strict";
+
+    var log = logger.getLogger("pat-display-time");
+    log.debug("pattern loaded");
+
+    var parser = new Parser("display-time");
 
-    var parser = new Parser('display-time');
     // input datetime options
-    parser.add_argument('format', '');
-    parser.add_argument('locale', '');
-    parser.add_argument('strict', false);
+    parser.add_argument("format", "");
+    parser.add_argument("locale", "");
+    parser.add_argument("strict", false);
 
     // output options
-    parser.add_argument('from-now', false);
-    parser.add_argument('output-format', '');
-    parser.add_argument('output-locale', '');
-
-
-    var displayTime = {
-        name: 'display-time',
-        trigger: '.pat-display-time',
-
-        init: function patDisplayTimeInit($el, opts) {
-            var options = parser.parse($el, opts);
-
-            if (options.strict === 'true') {
-                options.strict = true;
-            } else {
-                options.strict = false;
-            }
-
-            this.processDate($el, options);
+    parser.add_argument("from-now", false);
+    parser.add_argument("output-format", "");
+    parser.add_argument("output-locale", "");
+
+    return Base.extend({
+        name: "display-time",
+        trigger: ".pat-display-time",
+
+        init: function initUndefined () {
+            this.options = parser.parse(this.$el);
+            log.debug("pattern initialized");
+            this.processDate(this.$el, this.options);
         },
 
         processDate: function patDisplayTimeProcessDate($el, options) {
-            var dateStr = $el.attr('datetime');
```

Comment:
If you're using the Base pattern, then we have `$el` (and `options`) on the `this` parameter. So there's no need to pass them to `patDisplayTimeProcessDate`.

Instead, you can just get them from `this`, i.e. `this.$el` and `this.options`.